packages/eslint-plugin/tests/rules/array-type.test.ts

```diff
@@ -2192,6 +2192,22 @@ type BrokenArray = {
       'type T = ReadonlyArray<ReadonlyArray<string>>',
       'generic',
     );
+    // conditional types
+    testOutput(
+      'array',
+      'type Conditional<T> = Array<T extends string ? string : number>',
+      'type Conditional<T> = (T extends string ? string : number)[]',
+    );
+    testOutput(
+      'array-simple',
+      'type Conditional<T> = (T extends string ? string : number)[]',
+      'type Conditional<T> = Array<T extends string ? string : number>',
+    );
+    testOutput(
+      'generic',
+      'type Conditional<T> = (T extends string ? string : number)[]',
+      'type Conditional<T> = Array<T extends string ? string : number>',
+    );
```

Comment:
[Testing] The block [`array-type (nested)`](https://github.com/typescript-eslint/typescript-eslint/blob/08a8eff1ce06929d9a602fb1f14beaef69848488/packages/eslint-plugin/tests/rules/array-type.test.ts#L1963) is for nested array tests, these tests can be moved over to [the standard tests block](https://github.com/typescript-eslint/typescript-eslint/blob/d24a82854d06089cbd2a8801f2982fd4781f3701/packages/eslint-plugin/tests/rules/array-type.test.ts#L12-L1959).